### PR TITLE
To exclude the OldRelease LTP updation if an application don't have the forwarding PromptForEmbeddingCausesRequestForBequeathingData

### LINKED ISSUE #827: To exclude the OldRelease LTP updation if an application don't have the forwarding PromptForEmbeddingCausesRequestForBequeathingData
If an application(RequestSequenceRepresentation) don't have a OldRelease(that transfers data) , then PromptForEmbeddingCausesRequestForBequeathingData forwarding will not be there and the preceeding application LTP instance shall not be depicted(as there is no such instance). 
In such case , the current basic services , register-yourself and embed-yourself , where the forwarding "PromptForEmbeddingCausesRequestForBequeathingData"  is used to find the OldRelease instances to update the application-name , release-number , tcp information will not work and also not required. Logic shall be modified to consider applications without PromptForEmbeddingCausesRequestForBequeathingData(means , there is no OldRelease to perform bequeath-your-data-and-die) to not do the procedure to update OldRelease information.

### PR DESCRIPTION
**Pull request by PrathibaJee**
Fixes #827
@DanaSunal  is the reviewer.

### CODE REVIEW

File: server/basicServices/basicServices/BasicServicesService.js [hunk 1]
```diff
@@ -99,9 +99,12 @@ exports.embedYourself = function (body, user, originator, xCorrelator, traceIndi
       let isOldApplicationTcpClientUpdated = false;
       let oldApplicationTcpClientUuid;
       let oldApplicationForwardingTag = "PromptForEmbeddingCausesRequestForBequeathingData";
+      let isOldReleaseExist = await isForwardingNameExist(oldApplicationForwardingTag);
+      let oldApplicationNameInConfiguration;
+      if(isOldReleaseExist){
```
Comment (DanaSunal) [inline server/basicServices/basicServices/BasicServicesService.js:104]: Format the code please. Missing spaces and wrong indentation.

File: server/basicServices/basicServices/BasicServicesService.js [hunk 2]
```diff
@@ -1444,6 +1449,14 @@ exports.updateOperationKey = function (body, user, originator, xCorrelator, trac
   });
 }
 
+async function isForwardingNameExist(forwardingName) {
+  const forwardingConstruct = await ForwardingDomain.getForwardingConstructForTheForwardingNameAsync(forwardingName);
+  if (forwardingConstruct === undefined) {
```
Comment (DanaSunal) [inline server/basicServices/basicServices/BasicServicesService.js:1454]: return forwardingConstruct !== undefined;

File: server/basicServices/basicServices/services/PrepareForwardingAutomation.js [hunk 3]
```diff
@@ -123,8 +123,12 @@ exports.registerYourself = function (logicalTerminationPointconfigurationStatus,
             }
 
             registrationApplicationRequestBody.tcpServerList = tcpServerList;
-            registrationApplicationRequestBody.precedingApplicationName = oldApplicationName;
-            registrationApplicationRequestBody.precedingReleaseNumber = oldReleaseNumber;
+            if(oldApplicationName){
```
Comment (DanaSunal) [inline server/basicServices/basicServices/services/PrepareForwardingAutomation.js:126]: Missing spaces.

**Review by DanaSunal — CHANGES_REQUESTED** [review]

File: server/basicServices/basicServices/services/PrepareForwardingAutomation.js (hunk 3 above)
Comment (PrathibaJee) [reply server/basicServices/basicServices/services/PrepareForwardingAutomation.js:126]: Formatted the file.

File: server/basicServices/basicServices/BasicServicesService.js (hunk 1 above)
Comment (PrathibaJee) [reply server/basicServices/basicServices/BasicServicesService.js:104]: Formatted the entire file.

File: server/basicServices/basicServices/BasicServicesService.js (hunk 2 above)
Comment (PrathibaJee) [reply server/basicServices/basicServices/BasicServicesService.js:1454]: Thanks for proposing the optimization.

**Review by DanaSunal — APPROVED** [review]

**Pull request merged**